### PR TITLE
[Epic] Register and verify recovery email

### LINKED ISSUE #3145: [Recovery] Register and verify an email address via Settings
## What is the feature about

Allow users to register and verify an email address on the settings page in order to receive email notifications about pending recovery attempts.

## The list of requirements

- [ ] #3291
- [ ] #3320
- [ ] #3321

## Designs/sketches

- [Figma](https://www.figma.com/file/pO43pjVwR34Uy1ihMsY3OM/Recovery-Hub?type=design&node-id=1816-48897&mode=design&t=syybxe0VIclJTlmh-0)

## Links

Tech spec: [Notion](https://www.notion.so/safe-global/As-a-Safe-owner-I-want-to-register-an-email-address-so-that-I-can-receive-recovery-notifications-80f8336053694dec917db5d3072a6610)

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Resolves #3145

This PR will include the following features:

- [x] #3291 
- [x] #3320
- [x] #3321
- [x] #3146 
- [x] #3149

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "@safe-global/safe-core-sdk-utils": "^1.7.4",
     "@safe-global/safe-deployments": "1.32.0",
     "@safe-global/safe-ethers-lib": "^1.9.4",
-    "@safe-global/safe-gateway-typescript-sdk": "3.17.0-alpha.1",
+    "@safe-global/safe-gateway-typescript-sdk": "3.18.0-alpha.2",
```
Comment (katspaugh) [inline package.json:63]: You can now use 3.19.0

**Comment by katspaugh** [comment]
@usame-algan looks like emails are blocked on the backend side, so I'll close this PR for now. We can restore the branch if needed.

**Comment by usame-algan** [comment]
> @usame-algan looks like emails are blocked on the backend side, so I'll close this PR for now. We can restore the branch if needed.

Accidentally clicked delete branch 😅 Will check if I can recover

**Comment by katspaugh** [comment]
I restored it, no worries.

**Pull request closed without merging**